collector/receiver/telemetryapireceiver/config_test.go

```diff
@@ -15,11 +15,122 @@
 )
 
+func TestLoadConfig(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		id       component.ID
+		expected component.Config
+	}{
+		{
+			id:       component.NewID(component.MustNewType("telemetryapi")),
+			expected: NewFactory("extensionID").CreateDefaultConfig(),
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "1"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{platform, function, extension},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "2"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{platform},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "3"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{function},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "4"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{extension},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "5"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{platform, function},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "6"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{platform, extension},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "7"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{function, extension},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "8"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "9"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{function, extension},
+			},
+		},
+		{
+			id: component.NewIDWithName(component.MustNewType("telemetryapi"), "10"),
+			expected: &Config{
+				extensionID: "extensionID",
+				Port:        12345,
+				Types:       []string{function, extension},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.id.String(), func(t *testing.T) {
+			cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
+			require.NoError(t, err)
+			factory := NewFactory("extensionID")
+			cfg := factory.CreateDefaultConfig()
+			sub, err := cm.Sub(tt.id.String())
+			require.NoError(t, err)
+			require.NoError(t, sub.Unmarshal(cfg))
+			require.NoError(t, component.ValidateConfig(cfg))
+			require.Equal(t, tt.expected, cfg)
+		})
+	}
+}
+
```

Comment:
There's an approach in other Go tests in the project that would reduce some duplication and improve the specification of these tests. It looks more like this:

```
func TestLoadConfig(t *testing.T) {
    t.Parallel()

    // Helper function to create expected Config
    createExpectedConfig := func(types ...string) *Config {
        return &Config{
            extensionID: "extensionID",
            Port:        12345,
            Types:       types,
        }
    }

    tests := []struct {
        name     string
        id       component.ID
        expected component.Config
    }{
        {
            name:     "default",
            id:       component.NewID(component.MustNewType("telemetryapi")),
            expected: NewFactory("extensionID").CreateDefaultConfig(),
        },
        {
            name:     "all types",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "1"),
            expected: createExpectedConfig(platform, function, extension),
        },
        {
            name:     "platform only",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "2"),
            expected: createExpectedConfig(platform),
        },
        {
            name:     "function only",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "3"),
            expected: createExpectedConfig(function),
        },
        {
            name:     "extension only",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "4"),
            expected: createExpectedConfig(extension),
        },
        {
            name:     "platform and function",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "5"),
            expected: createExpectedConfig(platform, function),
        },
        {
            name:     "platform and extension",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "6"),
            expected: createExpectedConfig(platform, extension),
        },
        {
            name:     "function and extension",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "7"),
            expected: createExpectedConfig(function, extension),
        },
        {
            name:     "empty types",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "8"),
            expected: createExpectedConfig(),
        },
        {
            name:     "function and extension (alternative syntax)",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "9"),
            expected: createExpectedConfig(function, extension),
        },
        {
            name:     "function and extension (another syntax)",
            id:       component.NewIDWithName(component.MustNewType("telemetryapi"), "10"),
            expected: createExpectedConfig(function, extension),
        },
    }

    for _, tt := range tests {
        t.Run(tt.name, func(t *testing.T) {
            cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
            require.NoError(t, err)

            factory := NewFactory("extensionID")
            cfg := factory.CreateDefaultConfig()

            sub, err := cm.Sub(tt.id.String())
            require.NoError(t, err)
            require.NoError(t, sub.Unmarshal(cfg))
            require.NoError(t, component.ValidateConfig(cfg))
            
            require.Equal(t, tt.expected, cfg)
        })
    }
}

Reply:
Applied. Thanks for recommendation!